app/osparc.py

```diff
@@ -1,71 +1,159 @@
-from app.config import Config
 import json
 import osparc
 import tempfile
+
+from app.config import Config
+from flask import abort
+from osparc.rest import ApiException
 from time import sleep
 
 
+OPENCOR_SOLVER = "simcore/services/comp/opencor"
+DATASET_4_SOLVER = "simcore/services/comp/rabbit-ss-0d-cardiac-model"
+DATASET_17_SOLVER = "simcore/services/comp/human-gb-0d-cardiac-model"
+DATASET_78_SOLVER = "simcore/services/comp/kember-cardiac-model"
+
+
 class SimulationException(Exception):
     pass
 
 
-def run_simulation(model_url, json_config):
-    with tempfile.NamedTemporaryFile(mode="w+") as temp_config_file:
-        json.dump(json_config, temp_config_file)
+def start_simulation(data):
```

Comment:
I tried to think of a way to split this function but couldn't come up with a simple solution, not least since it relies on the use of exceptions which would need to be propagated.